drand/drand_validator_msg_test.go

```diff
@@ -24,3 +24,22 @@ func TestConstructCommitMessage(test *testing.T) {
 		test.Errorf("Commit message is not constructed in the correct size: %d", len(msg))
 	}
 }
+
+func TestProcessInitMessage(test *testing.T) {
+	leader := p2p.Peer{IP: "127.0.0.1", Port: "19999"}
+	validator := p2p.Peer{IP: "127.0.0.1", Port: "55555"}
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+	dRand.blockHash = [32]byte{}
+	msg := dRand.constructInitMessage()
+
+	if len(msg) != 93 {
```

Comment:
to check the length is not always reliable, any better way?

Reply:
I think a better way would be to unMarshall & check for the payload. Let me know your thoughts.

Reply:
make sense to me.